src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionQRScannerViewController.swift

```diff
@@ -46,13 +46,29 @@ protocol QRScannerViewController: class {
 	func present(_: UIViewController, animated: Bool, completion: (() -> Void)?)
 }
 
-final class ExposureSubmissionQRScannerNavigationController: UINavigationController {
-	var exposureSubmissionService: ExposureSubmissionService?
+final class ExposureSubmissionQRScannerNavigationController: UINavigationController, ExposureSubmissionCoordinatorViewController {
 
+	// MARK: - Attributes.
+	var coordinator: ExposureSubmissionCoordinator?
+	var exposureSubmissionService: ExposureSubmissionService?
```

Comment:
again, same question as above, do both need to be optional here?

Reply:
yes. Same question here, if some service is injected, we don't need to set it as optional.
I'll check out the branch to review it.

Reply:
Meanwhile, please check other code, if the injected service won't be re-assigned or weak reference, please just set it with `let`

Reply:
good point! I think in any case we should pass a non-optional reference to the constructor, however I will adapt the stored references in the view controllers to be `private weak var` to avoid possible reference cycles to the exposure submission service (that is kept alive by a strong reference in the coordinator)